storage-xray-udp/src/main/java/zipkin2/storage/xray_udp/UDPMessageEncoder.java

```diff
@@ -314,6 +319,15 @@ static void writeJson(Span span, Buffer buffer) throws IOException {
     writer.flush();
   }
 
+  static String getOrigin() {
+    return getenv("AWS_XRAY_ORIGIN", null);
```

Comment:
we typically defer env expansion to spring config. ex the yaml file and map to fields here.. That said, I don't think the collector will know the origin.

Reply:
Totally agree with your observation. It was a bit short sighted. I have since updated the code to not depend on env of storage module and instead map from `xray.origin` tag on the span.

Regarding spring mechanism for the env expansion since the encoder is statically referenced from the storage class it seemed more straightforward to get it directly.